mathesar/api/viewsets/tables.py

```diff
@@ -96,27 +99,36 @@ def previews(self, request, pk=None):
         table = self.get_object()
         serializer = TablePreviewSerializer(data=request.data)
         serializer.is_valid(raise_exception=True)
-        columns = serializer.data["columns"]
+        columns_field_key = "columns"
+        columns = serializer.data[columns_field_key]
 
         column_names = [col["name"] for col in columns]
         if not len(column_names) == len(set(column_names)):
-            raise ValidationError("Column names must be distinct")
+            raise CustomValidationError([ExceptionBody(ErrorCodes.DistinctColumnNameRequired.value,
```

Comment:
This is more involved than I think is necessary. It should be possible to just say something like:

```
raise DistinctColumnsRequiredException(message="Column names must be distinct")
```

The error code should be associated with the `DistinctColumnsRequiredException` in the `exceptions` module, it shouldn't need to be defined here.

This validation logic should ideally be in the `validate_columns` method of the serializer, not here ([see docs](https://www.django-rest-framework.org/api-guide/serializers/#validation)). Then it will automatically associate it with the `columns` field and we don't have to do it manually.

Reply:
> The error code should be associated with the DistinctColumnsRequiredException in the exceptions module, it shouldn't need to be defined here.

Yes that is the idea, I haven't added other custom Exceptions yet as it will be dealt with [in this issue](https://github.com/centerofci/mathesar/issues/883). But do you think it will be better to create a new Exception for one-off validations like this? I was looking at using the `message` or `error code` to match with the relevant information, but found that similar there are places where a different message is needed for a similar error code.

Reply:
> This validation logic should ideally be in the validate_columns method of the serializer, not here

I am aware of it, been itching to move it for a long it, but didn't want to pollute the PR, but it does seem like a good time to move them into serializers

Reply:
> Do you think it will be better to create a new Exception for one-off validations like this?

I think it will be easiest to have a single pattern to follow for tying error codes to exceptions, even for one-off exceptions.

It's fine to pass different `message` values to the same exception, you can do this for Python exceptions as well so it's a common pattern.

> it does seem like a good time to move them into serializers

Since you're already editing that particular code, I think it's fine to do it now. You don't have to do it for all of the code in this PR.